### ISSUE TITLE
Modifying Material sgnum doesn't update affected lattice parameters

### ISSUE THREAD

**Issue by psavery** [HEXRD/hexrd#205]
This was mentioned first in https://github.com/HEXRD/hexrdgui/issues/767#issue-815874490

Note the following test:
```python
from hexrd.material import Material
  
mat = Material()

def print_params():
    print('lattice Params:', *[x.value for x in mat.latticeParameters])

print(f'{mat.sgnum=}')
print_params()

mat.sgnum = 165

print(f'{mat.sgnum=}')
print_params()
```

Output:
```
mat.sgnum=225
lattice Params: 3.61 3.61 3.61 90.0 90.0 90.0
mat.sgnum=165
lattice Params: 3.61 3.61 3.61 90.0 90.0 90.0
```

You can see that even though the space group number was changed to 165 (a trigonal lattice type), the
angles all still remain 90, when the third angle should be 120. It looks like lattice parameters are not
updating when the space group number changes, and it is also affecting later calculations.

I guess we need to add some update to the lattice parameters in the [sgnum setter?](https://github.com/HEXRD/hexrd/blob/8917d038aaeef9e24c5d3d335e458bd7cf30b4f1/hexrd/material.py#L588-L599).

**Comment by saransh13** [comment]
Ok, I'm going to push a fix for this. Thanks for catching this @psavery

**Comment by joelvbernier** [comment]
I thought we had an issue for this already @saransh13?

**Comment by joelvbernier** [comment]
Yes, it is on hexrdgui: https://github.com/HEXRD/hexrdgui/issues/767

**Comment by joelvbernier** [comment]
Is there a way to link issues across repos @psavery or @cjh1?

**Comment by joelvbernier** [comment]
Adding a ref to this issue in the assoc hexrdgui one for now (comments).

**Comment by cjh1** [comment]
Yes, that is the closes you can do.

**Comment by joelvbernier** [comment]
added comments to HEXRD/hexrdgui#767

**Comment by saransh13** [comment]
Ok I think I've figured this one out. PR coming soon.

**Comment by psavery** [comment]
This was fixed by #207
